Use enums to indicate important/implicit in CSSPropertyParser functions

weinig opened the pull request:
#### eec6993f0edb4cb38c6614213ac822df7e19799b
<pre>
Use enums to indicate important/implicit in CSSPropertyParser functions
<a href="https://bugs.webkit.org/show_bug.cgi?id=287112">https://bugs.webkit.org/show_bug.cgi?id=287112</a>

Reviewed by NOBODY (OOPS!).

Replace bare booleans with single bit enums to make it clear where these
parameters are being passed.

* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.h:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eec6993f0edb4cb38c6614213ac822df7e19799b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88508 "Passed style check") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8027 "Failed to checkout and rebase branch from PR 40075") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42948 "Failed to checkout and rebase branch from PR 40075") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93466 "Failed to checkout and rebase branch from PR 40075") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39262 "Failed to checkout and rebase branch from PR 40075") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/90559 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8414 "Failed to checkout and rebase branch from PR 40075") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16211 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/93466 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/39262 "Failed to checkout and rebase branch from PR 40075") 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91510 "Build is in progress. Recent messages:OS: Sonoma (14.6), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Passed webkitperl tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/130/builds/8414 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/55/builds/42948 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/93466 "Failed to checkout and rebase branch from PR 40075") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/130/builds/8414 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/55/builds/42948 "Failed to checkout and rebase branch from PR 40075") | [❌ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38370 "Failed to checkout and rebase branch from PR 40075") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/130/builds/8414 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/55/builds/42948 "Failed to checkout and rebase branch from PR 40075") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95308 "Failed to checkout and rebase branch from PR 40075") | 
| | [❌ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15683 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/123/builds/16211 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/95308 "Failed to checkout and rebase branch from PR 40075") | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15939 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/55/builds/42948 "Failed to checkout and rebase branch from PR 40075") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/95308 "Failed to checkout and rebase branch from PR 40075") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18803 "Failed to checkout and rebase branch from PR 40075") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/55/builds/42948 "Failed to checkout and rebase branch from PR 40075") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8722 "Failed to checkout and rebase branch from PR 40075") | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15699 "Failed to checkout and rebase branch from PR 40075") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21007 "Failed to checkout and rebase branch from PR 40075") | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15440 "Failed to checkout and rebase branch from PR 40075") | | | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18889 "Failed to checkout and rebase branch from PR 40075") | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17222 "Failed to checkout and rebase branch from PR 40075") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/parser/CSSPropertyParser.cpp:
```diff
@@ -817,8 +817,8 @@ bool CSSPropertyParser::consumeFontVariantShorthand(bool important)
-    bool implicitLigatures = true;
-    bool implicitNumeric = true;
+    IsImplicit implicitLigatures = IsImplicit::Yes;
+    IsImplicit implicitNumeric = IsImplicit::Yes;
```
```suggestion
    auto implicitLigatures = IsImplicit::Yes;
    auto implicitNumeric = IsImplicit::Yes;
```

weinig committed:
Use enums to indicate important/implicit in CSSPropertyParser functions
https://bugs.webkit.org/show_bug.cgi?id=287112

Reviewed by NOBODY (OOPS!).

Replace bare booleans with single bit enums to make it clear where these
parameters are being passed.

* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.h:
